pkgs/by-name/li/libnick/package.nix

```
@@ -0,0 +1,51 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  cmake,
+  pkg-config,
+  boost,
+  curl,
+  glib,
+  libsecret,
+  maddy-markdown,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "libnick";
+  version = "2024.11.1";
+
+  src = fetchFromGitHub {
+    owner = "NickvisionApps";
+    repo = "libnick";
+    rev = "${finalAttrs.version}";
+    hash = "sha256-e64f8zG+/kV4BmG76UZAhDJL6IeDMbdXwyHgjYiSJtU=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    pkg-config
+  ];
+
+  buildInputs = [
+    boost
+    curl
+    glib
+    libsecret
+    maddy-markdown
+  ];
```

```suggestion
  buildInputs = [
    boost
    curl
    glib
    libsecret
    maddy-markdown
  ]
  ++ lib.optionals stdenv.hostPlatform.isUnix [
    glib
    openssl
  ]
  ++ lib.optional stdenv.hostPlatform.isWindows sqlcipher
  ++ lib.optional withLibsecret libsecret;
```
Here we can make some dependencies system specific, as well as satisfy the requirements for optional features